File: CashRegister-May2015/ShawnCampbellCPP/Denomination.cpp
//
// Created by Shawn Campbell on 4/28/15.
//

#include "Denomination.h"

void Denomination::setValue(int value) {
    this->value = value;
}

std::string Denomination::getDenomName() {
    return this->denom_name;
}

void Denomination::setDenomName(std::string name) {
    this->denom_name = name;
}

int Denomination::getValue() {
    return this->value;
}

Denomination::Denomination(std::string name, int value) {
    this->setDenomName(name);
    this->setValue(value);
}

File: CashRegister-May2015/ShawnCampbellSwift/CashRegister/CashRegister/Denomination.swift
//
// Created by Shawn Campbell on 5/18/15.
// Copyright (c) 2015 Team Awesome McPants. All rights reserved.
//
import Foundation

struct Denomination<T: Hashable, T2> {

    var denom = Dictionary<T, T2>()

    mutating func insert(value: T2, forKey key: T) -> Dictionary<T, T2>? {
        self.denom.removeAll()
        self.denom.updateValue(value, forKey: key)
        return denom
    }

    func get() -> T2? {
        return self.denom[self.denom.keys.array[0]]
    }
}

